### DISCUSSION TITLE
reinstalling MET for python embedding

### DISCUSSION THREAD

**Discussion by JosemariePalero** [dtcenter/METplus#2652] [Installation] [answered]
Hi. I would like to ask for assistance in reinstalling MET for python embedding. 

When I try to do python embedding, I get the following error:
DEBUG 3: Calling /home/met/miniconda3/envs/METplus/bin/python3 to run user's python script (/home/met/ai4rp/pythonTools/WRF-12kmtp_24hPE).
DEBUG 4: Writing temporary Python dataplane file:
DEBUG 4: 	/home/met/miniconda3/envs/METplus/bin/python3 /home/met/MET/MET-11.1.1/share/met/python/pyembed/write_tmp_dataplane.py /home/met/MET/METplus-5.1.0/Output/tmp/tmp_met_nc_2078319_0 /home/met/ai4rp/pythonTools/WRF-12kmtp_24hPE /share/mbs/DATA/WRF/raw/d01/2023/08/tp_24h/postwrf_d01_20230816_0000_f02400.gr2 APCP 20230817_0000 20230816_0000 24
Python Script:	'/home/met/MET/MET-11.1.1/share/met/python/pyembed/write_tmp_dataplane.py'
User Command:	'/home/met/ai4rp/pythonTools/WRF-12kmtp_24hPE /share/mbs/DATA/WRF/raw/d01/2023/08/tp_24h/postwrf_d01_20230816_0000_f02400.gr2 APCP 20230817_0000 20230816_0000 24'
Temporary File:	'/home/met/MET/METplus-5.1.0/Output/tmp/tmp_met_nc_2078319_0'
DEBUG 3: Initializing MET compile time python instance: /usr/bin/python3
DEBUG 4: Reading temporary Python dataplane file: /home/met/MET/METplus-5.1.0/Output/tmp/tmp_met_nc_2078319_0
Traceback (most recent call last):
  File "/home/met/MET/MET-11.1.1/share/met/python/met/dataplane.py", line 3, in <module>
    import numpy as np
ModuleNotFoundError: No module named 'numpy'

I know from these lines in my log that the python being used by the python embedding and the python that MET is using are not the same. Hence, the need to recompile MET. But when I try to recompile MET, the required library JASPER is not compiling.
Here's my compile_MET_all.log:

[compile_MET_all.log](https://github.com/user-attachments/files/16620560/compile_MET_all.log)

And here's my jasper.make.log:
[jasper.make.log](https://github.com/user-attachments/files/16620563/jasper.make.log)

Thank you very much.
For your reference, here's my install_met_env.METplus:
export CC=/home/met/miniconda3/envs/METplus/bin/gcc
export CXX=/home/met/miniconda3/envs/METplus/bin/g++
export FC=/home/met/miniconda3/envs/METplus/bin/gfortran
export F77=/home/met/miniconda3/envs/METplus/bin/gfortran
export F90=/home/met/miniconda3/envs/METplus/bin/gfortran
export TEST_BASE=/home/met/MET/MET-11.1.1
export COMPILER=gnu_14.1.0
export MET_SUBDIR=${TEST_BASE}
export MET_TARBALL=v11.1.1.tar.gz
export USE_MODULES=FALSE
export MET_PYTHON=/home/met/miniconda3/envs/METplus
export MET_PYTHON_BIN_EXE=/home/met/miniconda3/envs/METplus/bin/python3
export MET_PYTHON_CC='-I/home/met/miniconda3/envs/METplus/include/python3.12'
export MET_PYTHON_LD='-L/home/met/miniconda3/envs/METplus/lib/python3.12/config-3.12-x86_64-linux-gnu -L/home/met/miniconda3/envs/METplus/lib -lpython3.12 -lpthread -ldl  -lutil -lm'
export SET_D64BIT=FALSE

**Comment by jprestop** [comment]
Hi @JosemariePalero.  

Regarding the Python problem, it would be helpful to see the configuration file for compile_MET_all.sh, along with the config.log file in your MET-11.1.1 directory.  Could you please send those along?  You could also try setting an environment variable MET_PYTHON_EXE to point to the full path to and including the executable python3 file and then try rerunning and see if that helps.

Regarding the Jasper problem, I haven't encountered this problem before, but I did notice that you are using a newer compiler than I have used (GNU 14.1.0), which could be problematic with this older version of jasper.  I'd suggest updating the version of the jasper library that you are using and trying to compile it instead.  I have attached an updated version of compile_MET_all.sh (called compile_MET_all.new.sh in a zip format) and a newer version of the jasper library for you to try to install (this is the version of the jasper library that we will suggest using in our next official release).  Please add this jasper library package to your tar_files subdirectory.  You will need to have access to `cmake` in order to install it.  Please try running compile_MET_all.new.sh and let us know how it goes.

[compile_MET_all.new.sh.zip](https://github.com/user-attachments/files/16628026/compile_MET_all.new.sh.zip)
[jasper-2.0.25.tar.gz](https://github.com/user-attachments/files/16628027/jasper-2.0.25.tar.gz)

**Reply by JosemariePalero** [reply]
Thanks, I'll try that today and update you on the results.

**Reply by JosemariePalero** [reply]
Hi, regarding the Jasper problem, I have successfully compiled Jasper using the compile_MET_all.new.sh file and jasper-2.0.25.tar.gz you provided but had an error in compiling G2CLIB after that. So I think you're right that the latest compiler is incompatible with the older versions of the libraries needed. I think that in order to compile MET and its required libraries, I need to either update all tar files or downgrade my compiler. Thank you very much for your help. I think this is the answer I need.

**Reply by jprestop** [reply, marked as answer]
Hi @JosemariePalero. we do development and testing using Debian Linux. And we’re using Debian 12 (bookworm) which is the [latest stable release](https://www.debian.org/releases/) and uses [GCC version 12.2.0](https://packages.debian.org/bookworm/gcc). 

The next Debian release ([trixie](https://www.debian.org/releases/trixie/)) is slated to use [GCC version 14.1.0](https://packages.debian.org/trixie/gcc), but trixie is still in testing mode. We will eventually upgrade from bookworm to trixie at some point in the future, but certainly not before trixie moves from testing mode to the stable version, and that usually takes a long time as issues with newer compilers are shaken out and resolved.

Because of this, using an earlier GNU compiler, like 12.2.0, is what we recommend.

**Reply by JosemariePalero** [reply]
Thank you very much for your recommendation.

**Reply by jprestop** [reply]
@JosemariePalero  You're welcome. If you don't have any further questions in this Discussion at this time, please go ahead and click the "Mark as answer" button. Then, I will LOCK this conversation to prevent future posts. Our team has decided that once discussions have been answered, and the answer has been confirmed, we’ll lock them. We want to encourage users to ask new questions in new discussions rather than posting to old ones. Hopefully that’ll make the questions/answers easier for other users to follow. So if/when more issues/questions arise, please feel free to start a new discussion. I do see that you have started another Discussion already and I will take a look at that soon.
